src/PointwiseFunctions/AnalyticSolutions/Elasticity/BentBeam.hpp

```diff
@@ -4,26 +4,60 @@
 #pragma once
 
 #include <limits>
+#include <pup.h>
 
-#include "DataStructures/DataBox/Prefixes.hpp"   // IWYU pragma: keep
-#include "DataStructures/Tensor/Tensor.hpp"      // IWYU pragma: keep
-#include "Elliptic/Systems/Elasticity/Tags.hpp"  // IWYU pragma: keep
+#include "DataStructures/CachedTempBuffer.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Elliptic/Systems/Elasticity/Tags.hpp"
 #include "Options/Options.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Elasticity/AnalyticSolution.hpp"
 #include "PointwiseFunctions/Elasticity/ConstitutiveRelations/IsotropicHomogeneous.hpp"
-#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/Tags.hpp"  // IWYU pragma: keep
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/Registration.hpp"
 #include "Utilities/TMPL.hpp"
 #include "Utilities/TaggedTuple.hpp"
 
-// IWYU pragma: no_forward_declare Tensor
+namespace Elasticity::Solutions {
+
+namespace detail {
+template <typename DataType>
+struct BentBeamVariables {
```

Comment:
This detail class seems unnecessary -- why use this instead of just creating template specializations of the `variables` function? I would have thought that supporting the retrieval of solutions via the `variables` function was the entire point of the `dynamic_cast` trickery in the base class.

Reply:
The issue is the template on registrars. That prevents moving the implementations to the cpp file. If you have an idea how to deal with that I'd be happy to hear 😄

Reply:
Can you clarify why you need to use the registrars trick here? I would have thought that, for instance, for the Poisson analytic solutions, you could always just explicitly specify all of the Poisson analytic solutions in its `creatable_classes`?

Reply:
Hm, that would mean the executable can't choose which analytic solution or data classes it allows, but it's always the full list of available classes. I'll have to think if that can work in the future. For more involved systems like XCTS I'll likely have more than one "solve this system" executables because they need to do more things, e.g. a neutron-star solve needs to interpolate data to the center of the star in every nonlinear step and a vacuum solve may want to find apparent horizons instead. This might be solvable with more base classes.. I'll think about it a bit more :)